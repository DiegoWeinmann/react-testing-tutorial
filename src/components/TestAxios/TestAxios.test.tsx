import React from 'react';
import {
  render,
  cleanup,
  fireEvent,
  waitForElement
} from '@testing-library/react';
import TestAxios from './TestAxios';
import axiosMock from 'axios';

jest.mock('axios');
axiosMock.get = jest.fn().mockResolvedValue({
  data: {
    greeting: 'hello world'
  }
});

afterEach(cleanup);

const renderUi = () => {
  return render(<TestAxios url={`:`} />);
};

describe('<TestAxios/>', () => {
  it('matches the snapshot.', () => {
    const { asFragment } = renderUi();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display a loading text.', () => {
    const { getByTestId } = renderUi();
    expect(getByTestId('loading')).toHaveTextContent('Loading...');
  });

  it('should load and display the data', async () => {
    const url = '/greeting';
    const { getByTestId } = render(<TestAxios url={url} />);

    fireEvent.click(getByTestId('fetch-data'));

    const greetingData = await waitForElement(() => getByTestId('show-data'));

    expect(axiosMock.get).toHaveBeenCalledTimes(1);
    expect(axiosMock.get).toHaveBeenCalledWith(url);
    console.log(greetingData);
    expect(greetingData).toHaveTextContent('hello world');
  });
});
