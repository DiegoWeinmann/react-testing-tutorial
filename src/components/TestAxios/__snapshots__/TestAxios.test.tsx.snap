// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TestAxios/> matches the snapshot. 1`] = `
<DocumentFragment>
  <button
    data-testid="fetch-data"
  >
    Load Data
  </button>
  <h1
    data-testid="loading"
  >
    Loading...
  </h1>
</DocumentFragment>
`;
